packages/api/amplify_api/lib/src/graphql/send_graphql_request.dart

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'dart:convert';
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:http/http.dart' as http;
+import 'package:meta/meta.dart';
+
+import '../util.dart';
+import 'graphql_response_decoder.dart';
+
+/// Converts the [GraphQLRequest] to an HTTP POST request and sends with ///[client].
+@internal
+Future<GraphQLResponse<T>> sendGraphQLRequest<T>({
+  required GraphQLRequest<T> request,
+  required http.Client client,
+  required Uri uri,
+}) async {
+  try {
+    final body = {'variables': request.variables, 'query': request.document};
+    final graphQLResponse = await client.post(uri, body: json.encode(body));
+
+    final responseBody = json.decode(graphQLResponse.body);
+
+    if (responseBody is! Map<String, dynamic>) {
+      throw ApiException(
+          'unable to parse GraphQLResponse from server response which was not a JSON object.',
+          underlyingException: graphQLResponse.body);
+    }
+
+    final responseData = responseBody['data'];
+    // Preserve `null`. json.encode(null) returns "null" not `null`
+    final responseDataJson =
+        responseData != null ? json.encode(responseData) : null;
```

Comment:
callout (no change needed in this PR): Looks like we are decoding the response from string to map, then re-encoding part of it to a string to pass to decoder, which then converts it back to JSON again. Looks like we are doing this bc we are basing it off the decoder which was built around the method channel implementation that returns a string. We should look into changing the decoder to accept a map so that we only have to decode it once.